### ISSUE TITLE
to_sql sqlalchemy pickle error with 2022.2.0

### ISSUE THREAD

**Issue by mgsnuno** [dask/dask#8738]
**What happened**:
to_sql sqlalchemy pickle error with dask 2022.2.0

**What you expected to happen**:
to_sql produces no pickle error with 2022.1.1

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import dask.dataframe as dd
from dask.distributed import Client

client = Client()

data = dd.from_pandas(pd.DataFrame([1, 2, 3, 4]), chunksize=20000)
data.to_sql(
    "testtest",
    uri=str(sqlengine.url),
    if_exists="replace",
    schema=None,
    index=False,
)
```

<details>
<summary>Traceback</summary>

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/distributed/worker.py", line 4442, in dumps_function
    result = cache_dumps[func]
  File "/opt/conda/lib/python3.9/site-packages/distributed/utils.py", line 1360, in __getitem__
    value = super().__getitem__(key)
  File "/opt/conda/lib/python3.9/collections/__init__.py", line 1058, in __getitem__
    raise KeyError(key)
KeyError: <function to_sql.<locals>.make_meta at 0x7f05284b3af0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/distributed/protocol/pickle.py", line 49, in dumps
    result = pickle.dumps(x, **dump_kwargs)
AttributeError: Can't pickle local object 'to_sql.<locals>.make_meta'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.9/site-packages/dask/dataframe/core.py", line 1580, in to_sql
    return to_sql(
  File "/opt/conda/lib/python3.9/site-packages/dask/dataframe/io/sql.py", line 850, in to_sql
    dask.compute(result)
  File "/opt/conda/lib/python3.9/site-packages/dask/base.py", line 573, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/distributed/client.py", line 2974, in get
    futures = self._graph_to_futures(
  File "/opt/conda/lib/python3.9/site-packages/distributed/client.py", line 2882, in _graph_to_futures
    dsk = dsk.__dask_distributed_pack__(self, keyset, annotations)
  File "/opt/conda/lib/python3.9/site-packages/dask/highlevelgraph.py", line 1054, in __dask_distributed_pack__
    "state": layer.__dask_distributed_pack__(
  File "/opt/conda/lib/python3.9/site-packages/dask/highlevelgraph.py", line 425, in __dask_distributed_pack__
    dsk = toolz.valmap(dumps_task, dsk)
  File "cytoolz/dicttoolz.pyx", line 178, in cytoolz.dicttoolz.valmap
  File "cytoolz/dicttoolz.pyx", line 203, in cytoolz.dicttoolz.valmap
  File "/opt/conda/lib/python3.9/site-packages/distributed/worker.py", line 4480, in dumps_task
    return {"function": dumps_function(task[0]), "args": warn_dumps(task[1:])}
  File "/opt/conda/lib/python3.9/site-packages/distributed/worker.py", line 4444, in dumps_function
    result = pickle.dumps(func, protocol=4)
  File "/opt/conda/lib/python3.9/site-packages/distributed/protocol/pickle.py", line 60, in dumps
    result = cloudpickle.dumps(x, **dump_kwargs)
  File "/opt/conda/lib/python3.9/site-packages/cloudpickle/cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)
  File "/opt/conda/lib/python3.9/site-packages/cloudpickle/cloudpickle_fast.py", line 602, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle 'sqlalchemy.cprocessors.UnicodeResultProcessor' object
```
</details>

**Anything else we need to know?**:
tested with SQLAlchemy 1.4.31

**Environment**:

- Dask version: 2022.2.0
- Python version: 3.9.10
- Operating System: linux
- Install method (conda, pip, source): conda/mamba

**Comment by mgsnuno** [comment]
@jsignell I think it might have to do with this change: https://github.com/dask/dask/blob/f634f2b01caeb10dcd33734f780d9bb6056a21d9/dask/dataframe/io/sql.py#L798

the `engine` is created already at this stage and the engine is not picklable

**Comment by jsignell** [comment]
Thanks for raising this @mgsnuno! I think you are right in your diagnosis. I guess our options are to revert that #8609 or figure out a way to pickle the engine object. Ping @scharlottej13 and @ian-r-rose for options on the best path forward.

**Comment by ian-r-rose** [comment]
Ah, yeah, we shouldn't be trying to create the `engine` directly at graph construction time for exactly this reason. The `read_sql` path defers engine creation until `_read_sql_chunk`, this `to_sql` be doing something similar.

It should be a reasonably straightforward fix, though perhaps we should revert #8609 until we can get it in.

**Comment by jsignell** [comment]
Yeah it does seem like we could write a similar `_to_sql_chunk`.
